source/features/show-github-action-next-time.tsx

```diff
@@ -0,0 +1,71 @@
+import './show-github-action-next-time.css';
+import React from 'dom-chef';
+import cache from 'webext-storage-cache';
+import select from 'select-dom';
+import elementReady from 'element-ready';
+import * as pageDetect from 'github-url-detection';
+import {parseCron} from '@cheap-glitch/mi-cron';
+
+import features from '.';
+import * as api from '../github-helpers/api';
+import {getRepoURL, getRepoGQL, getCurrentBranch} from '../github-helpers';
+
+const getWorkflows = cache.function(async (): Promise<Array<{[index: string]: {[index: string]: string}}>> => {
+	const {repository: {object: {entries: workflows}}} = await api.v4(
+		`repository(${getRepoGQL()}) {
+			object(expression: "${getCurrentBranch()}:.github/workflows") {
+				... on Tree {
+					entries {
+						object { ... on Blob { text } }
+					}
+				}
+			}
+		}
+	`);
+
+	return workflows;
+}, {
+	maxAge: {
+		days: 1
+	},
+	cacheKey: () => __filebasename + ':' + getRepoURL()
+});
+
+function parseYamls(yamls: string[]): {[index: string]: string} {
+	return Object.fromEntries(yamls
+		.map(yaml => [(/^name:\s+(\S+)$/m.exec(yaml) ?? [])[1], (/^\s*-\scron:\s+['"](.+)['"]/m.exec(yaml) ?? [])[1]])
+		.filter(([name, cron]) => name !== undefined && cron !== undefined)
```

Comment:
This double loop with long lines is unreadable, it’s best to use a regular “for of”